lua/apisix/core/utils.lua

```diff
@@ -83,12 +83,28 @@ function _M.dns_parse(resolvers, domain)
 end
 
 
+local function rfind_char(s, ch, idx)
```

Comment:
The `rfind_char` (reverse find a char) is critical for the correctness when handling the IPv6 address, because the IPv6 address itself might contains one or many colons.
As for performance, look up single character one by one when the port part is small might be as fast as a `memchr` call, which is used by LuaJIT under the hook.
The glibc's `memchr` implementation, https://repo.or.cz/glibc.git/blob/HEAD:/string/memchr.c#l67, does the same thing when the len(s) is small.

Reply:
got it, thx